### PR TITLE
Add QGamepad based input driver.

### PR DESCRIPTION
**Pull request by t-paul**

### CODE REVIEW

File: openscad.pro [hunk 1]
```diff
@@ -554,6 +554,15 @@ unix:!macx {
   SOURCES += src/input/JoystickInputDriver.cc
 }
 
+contains(QT_VERSION, ^5\\..*) {
```
Comment (kintel) [inline openscad.pro]: You can use `!lessThan(QT_MAJOR_VERSION, 5)` to be more future proof

File: src/input/QGamepadInputDriver.h [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+public:
+    QGamepadInputDriver();
+    virtual ~QGamepadInputDriver();
+    virtual void run();
+    virtual bool open();
+    virtual void close();
+
+    virtual const std::string & get_name() const;
+private:
+    QGamepad *gamepad;
```
Comment (kintel) [inline src/input/QGamepadInputDriver.h]: We could protect this object with a `unique_ptr` to ensure that this object gets deleted in all cases.

File: src/input/QGamepadInputDriver.h [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    char name[1024];
+    volatile bool stopRequest;
```
Comment (kintel) [inline src/input/QGamepadInputDriver.h]: What does this member variable do?

File: src/input/QGamepadInputDriver.h [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    QGamepadInputDriver();
+    virtual ~QGamepadInputDriver();
+    virtual void run();
+    virtual bool open();
+    virtual void close();
+
+    virtual const std::string & get_name() const;
+private:
+    QGamepad *gamepad;
+    char name[1024];
```
Comment (kintel) [inline src/input/QGamepadInputDriver.h]: This looks unused

File: src/input/QGamepadInputDriver.h (hunk 3 above)
Comment (MichaelPFrey) [reply src/input/QGamepadInputDriver.h]: stopRequest looks like an artefact from JoystickInputDriver.

The JoystickInputDriver uses the bool stopRequest to stop its main while loop.
As the while loop is in a different thread [the while loop is in the run method], I used volatile to make sure that the compiler would not start doing funny business while optimizing.

Given that the run methode of this driver is empty, there should be no need for this volatile bool.

Comment (t-paul) [reply src/input/QGamepadInputDriver.h]: Yep, both variables are just left-overs, I'm just going to remove those...

**Pull request merged**